### ISSUE TITLE
Rule IDs may not be unique

### ISSUE THREAD

**Issue by SolarBear** [alexshelkov/SimpleAcl#7]
I stumbled on a weird problem which happened only on my machine. It's working alright for all of my coworkers, as far as I'm aware.

We create a bunch of rules (here's my edited code) :
```php
        $this->addRule(
            new Role(self::ROLE1),
            new Resource(self::RESOURCE1),
            new Rule(self::RULE1),
            $value1
        );

        $this->addRule(
            new Role(self::ROLE2),
            new Resource(self::RESOURCE2),
            new Rule(self::RULE2),
            $value1
        );
//and so on
```

Our code creates 6 such rules and I ended up having anywhere from 1 to 3 rules, the number would change every time I refreshed.

The problem is in `SimpleAcl\Acl.php` in the `hasRule` method : my `Rule` objects have the same ids! I found out that, by default, ids have a value generated by the `uniqid` core function and this function will return an ID that depends on the current time in microseconds. It would appear that, on my machine (running PHP 5.5.12 on Windows 7 x64) these uniqids are generated in the same microsecond and thus have the same id. For what it's worth, setting an ID manually and adding extra entropy fixes my problem:

```php
        $rule = new Rule(self::RULE1);
        $rule->setId(uniqid('', true));

        $this->addRule(
            new Role(self::ROLE1),
            new Resource(self::RESOURCE1),
            $rule,
            $value1
        );
        //etc.
```

**Comment by alexshelkov** [comment]
Could you try to test it with this code:
```php
    /**
     * Creates an id for rule.
     *
     * @return mixed
     */
    protected function generateId()
    {
        return bin2hex(openssl_random_pseudo_bytes(10));
    }
```

**Comment by alexshelkov** [comment]
Closed in 2.0.22, now instead of uniqid use openssl_random_pseudo_bytes to generate id.

**Comment by SolarBear** [comment]
Yes, sorry for not replying, it's working fine now. Thanks!
